test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs

```diff
@@ -4768,6 +4768,693 @@ public Task TestPacManGetUnrestoredPackagesByDependencyOrder()
             Assert.Equal(bVersionRange.ToString(), packageDependency.VersionRange.ToString());
         }
 
+        [Fact]
+        public async Task TestPacManInstallPackage_BatchEvent()
+        {
+            // Arrange
+            var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateV3OnlySourceRepositoryProvider();
```

Comment:
These tests should use local packages instead of online packages. I know many other tests do this, but we're trying to move away from that since there is no benefit for tests like this to go online. 

See the PackagePreFetcherTests and InstallationCompatibilityTests for examples of adding test packages to local folders, it should be a simple change over.